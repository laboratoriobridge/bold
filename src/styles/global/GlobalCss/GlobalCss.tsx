import { Global as EmotionGlobal, Interpolation } from '@emotion/core/'
import React from 'react'

import { useTheme } from '../../hooks'
import { Theme } from '../../theme/createTheme'

export const GlobalCss = EmotionGlobal

export const DefaultGlobalCss = () => {
  const theme = useTheme()

  return <GlobalCss styles={createGlobalStyles(theme)} />
}

export const createGlobalStyles = ({ pallete, typography }: Theme): Interpolation => ({
  html: {
    fontSize: typography.sizes.html,
    boxSizing: 'border-box',
  },
  '*, *:before, *:after': {
    boxSizing: 'inherit',
  },
  body: {
    margin: 0,
    backgroundColor: pallete.surface.background,
    color: pallete.text.main,
    fontFamily: typography.fontFamily,
    fontSize: typography.sizes.text,
    lineHeight: typography.lineHeight,
    overflowY: 'scroll',
  },
  'button, input, optgroup, select, textarea': {
    /* Input elements do not inherit body's font styles */
    fontFamily: typography.fontFamily,
    fontSize: typography.sizes.text,
    lineHeight: 'inherit',
  },
  hr: {
    backgroundColor: pallete.divider,
    border: 'none',
    height: 1,
    margin: '1rem 0',
  },
  h1: { margin: 0, fontSize: '1.5rem' },
  h2: { margin: 0, fontSize: '1.25rem' },
  h3: { margin: 0, fontSize: '1rem' },
  h4: { margin: 0, fontSize: '0.875rem' },
  h5: { margin: 0, fontSize: '0.8125rem' },
  h6: { margin: 0, fontSize: '0.75rem' },
  p: {
    margin: '0',
    lineHeight: 1.5,
  },
  a: {
    cursor: 'pointer',
    color: pallete.primary.main,
    textDecoration: 'underline',
    fontWeight: 'bold',
    '&:hover': {
      textDecoration: 'none',
    },
  },
  mark: {
    background: pallete.highlight,
  },
})
