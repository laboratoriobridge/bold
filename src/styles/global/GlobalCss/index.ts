export { GlobalCss, DefaultGlobalCss } from './GlobalCss'
