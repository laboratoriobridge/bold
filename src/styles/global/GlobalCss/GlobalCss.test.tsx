import { sheet } from 'emotion'
import React from 'react'
import { render } from 'react-testing-library'

import { DefaultGlobalCss, GlobalCss } from './GlobalCss'

describe('GlobalCss', () => {
  it('should inject global CSS styles', () => {
    render(<GlobalCss styles={{ body: { color: 'red' } }} />)
    expect(sheet.container).toMatchSnapshot()
  })
})

describe('DefaultGlobalCss', () => {
  it('should inject default CSS styles from theme', () => {
    render(<DefaultGlobalCss />)
    expect(sheet.container).toMatchSnapshot()
  })
})
