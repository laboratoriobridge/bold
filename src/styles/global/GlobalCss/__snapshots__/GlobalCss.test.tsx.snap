// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultGlobalCss should inject default CSS styles from theme 1`] = `
<head>
  <style
    data-emotion="css"
  >
    
    @-webkit-keyframes animation-1scjtrl{from{-webkit-transform:rotate(0deg);-ms-transform:rotate(0deg);transform:rotate(0deg);}to{-webkit-transform:rotate(359deg);-ms-transform:rotate(359deg);transform:rotate(359deg);}}
  </style>
  <style
    data-emotion="css"
  >
    
    @keyframes animation-1scjtrl{from{-webkit-transform:rotate(0deg);-ms-transform:rotate(0deg);transform:rotate(0deg);}to{-webkit-transform:rotate(359deg);-ms-transform:rotate(359deg);transform:rotate(359deg);}}
  </style>
  <style
    data-emotion="css-global"
  >
    
    html{font-size:16px;box-sizing:border-box;}
  </style>
  <style
    data-emotion="css-global"
  >
    
    *,*:before,*:after{box-sizing:inherit;}
  </style>
  <style
    data-emotion="css-global"
  >
    
    body{margin:0;background-color:#F2F2F7;color:#2D343B;font-family:"IBM Plex Sans",sans-serif;font-size:0.8125rem;line-height:1.5;overflow-y:scroll;}
  </style>
  <style
    data-emotion="css-global"
  >
    
    button,input,optgroup,select,textarea{font-family:"IBM Plex Sans",sans-serif;font-size:0.8125rem;line-height:inherit;}
  </style>
  <style
    data-emotion="css-global"
  >
    
    hr{background-color:#DCDCE0;border:none;height:1px;margin:1rem 0;}
  </style>
  <style
    data-emotion="css-global"
  >
    
    h1{margin:0;font-size:1.5rem;}
  </style>
  <style
    data-emotion="css-global"
  >
    
    h2{margin:0;font-size:1.25rem;}
  </style>
  <style
    data-emotion="css-global"
  >
    
    h3{margin:0;font-size:1rem;}
  </style>
  <style
    data-emotion="css-global"
  >
    
    h4{margin:0;font-size:0.875rem;}
  </style>
  <style
    data-emotion="css-global"
  >
    
    h5{margin:0;font-size:0.8125rem;}
  </style>
  <style
    data-emotion="css-global"
  >
    
    h6{margin:0;font-size:0.75rem;}
  </style>
  <style
    data-emotion="css-global"
  >
    
    p{margin:0;line-height:1.5;}
  </style>
  <style
    data-emotion="css-global"
  >
    
    a{cursor:pointer;color:#0A67D1;-webkit-text-decoration:underline;text-decoration:underline;font-weight:bold;}
  </style>
  <style
    data-emotion="css-global"
  >
    
    a:hover{-webkit-text-decoration:none;text-decoration:none;}
  </style>
  <style
    data-emotion="css-global"
  >
    
    mark{background:#FFED94;}
  </style>
</head>
`;

exports[`GlobalCss should inject global CSS styles 1`] = `
<head>
  <style
    data-emotion="css-global"
  >
    
    body{color:red;}
  </style>
</head>
`;
